src/Features/Core/Portable/EditAndContinue/CommittedSolution.cs

```diff
@@ -136,55 +136,73 @@ public async Task<(Document? Document, DocumentState State)> GetDocumentAndState
+                case DocumentState.MatchesBuildOutput:
+                    // Note: committedDocument is null if we previously validated that a document that is not in
+                    // the committed solution is also not in the PDB. This means the document has been added during debugging.
+                    return (committedDocument, documentState);
+
+                case DocumentState.DesignTimeOnly:
+                    return (null, documentState);
+
+                case DocumentState.OutOfSync:
+                    if (reloadOutOfSyncDocument)
                     {
-                        case DocumentState.MatchesBuildOutput:
-                            // Note: committedDocument is null if we previously validated that a document that is not in
-                            // the committed solution is also not in the PDB. This means the document has been added during debugging.
-                            return (committedDocument, documentState);
+                        break;
+                    }
 
-                        case DocumentState.DesignTimeOnly:
-                            return (null, documentState);
+                    return (null, documentState);
 
-                        case DocumentState.OutOfSync:
-                            if (reloadOutOfSyncDocument)
-                            {
-                                break;
-                            }
+                case DocumentState.Indeterminate:
+                    // Previous attempt resulted in a read error. Try again.
+                    break;
 
-                            return (null, documentState);
+                case DocumentState.None:
+                    // Have not seen the document before, the document is not in the solution, or the document is source generated.
 
-                        case DocumentState.Indeterminate:
-                            // Previous attempt resulted in a read error. Try again.
-                            break;
+                    if (committedDocument == null)
+                    {
+                        var sourceGeneratedDocument = await solution.GetSourceGeneratedDocumentAsync(documentId, cancellationToken).ConfigureAwait(false);
+                        if (sourceGeneratedDocument != null)
+                        {
+                            // source generated files are never out-of-date:
+                            return (sourceGeneratedDocument, DocumentState.MatchesBuildOutput);
+                        }
 
-                        case DocumentState.None:
-                            throw ExceptionUtilities.Unreachable;
+                        // The current document is source-generated therefore the corresponding one is not present in the base solution.
+                        if (currentDocument is SourceGeneratedDocument)
+                        {
+                            return (null, DocumentState.MatchesBuildOutput);
+                        }
```

Comment:
This the case where the generator is now producing one more generated file than it didn't before?

Reply:
Yes. It's a newly added source-generated document.